### ISSUE TITLE
irregular image-channels raises RuntimeError

### ISSUE THREAD

**Issue by andreasoie** [taesungp/contrastive-unpaired-translation#149]
In the class UnalignedDataset(BaseDataset), you're hardcasting images to RGB on line 58-59, but if we'are training with color images in trainA and greyscale images in trainB (by using the --output_nc 1 in train.py) images are loaded from the dataset-loader with three-color channels regardless? 

```python
# e.g. python train.py  ........ --output_nc 1 --batch_size 4

for i, data in enumerate(dataset):  # inner loop within one epoch
    # data['A'].shape is then torch.Size([4, 3, 256, 256])
    # data['B'].shape is then torch.Size([4, 3, 256, 256])
    ...
```

**Comment by taesungp** [comment]
I apologize that this greyscale flag is not supported very well. Would you be able to just train the greyscale image as a special case of an RGB image, where all channels have duplicate values?

**Comment by andreasoie** [comment]
Yes, it's possible to train using --output_nc 3, where all channels are greyscale. I've read both the papers you've wrote (for CycleGan and CUT), and tested both models. CycleGAN works with input_nc = 3 and output_nc = 1 by setting lambda_idenitity to zero. For supporting training CUT using output_nc = 1, do you think a lot of code needs to be rewritten?
